src/main/java/uk/gov/hmcts/darts/task/service/impl/AutomatedTaskServiceImpl.java

```diff
@@ -0,0 +1,232 @@
+            cronExpression = automatedTaskEntity.get().getCronExpression();
+            log.debug("Task: {} cron expression: {}", taskName, cronExpression);
+        } else {
+            throw new DartsApiException(FAILED_TO_FIND_AUTOMATED_TASK);
+        }
+        return cronExpression;
+    }
+
+    /**
+     * Updates the automated task cron expression or throws DartsApiException.
+     * Calling this without the automated task being cancelled first, and then reloaded means the cron expression will only get picked up
+     * after the next execution once the automated task has run and then calculates the next execution time. Cancelling
+     * the automated task first means it will use the given cronExpression if the cron expression is valid.
+     * @param taskName name of automated task
+     * @param cronExpression cron expression
+     * @return true if the automated task is successfully updated
+     */
+    @Override
+    public boolean updateAutomatedTaskCronExpression(String taskName, String cronExpression) {
+        Optional<AutomatedTaskEntity> automatedTaskEntity = getAutomatedTaskEntityByTaskName(taskName);
+        if (automatedTaskEntity.isPresent()) {
+            AutomatedTaskEntity automatedTask = automatedTaskEntity.get();
+            if (CronExpression.isValidExpression(cronExpression)) {
+                automatedTask.setCronExpression(cronExpression);
+                automatedTaskRepository.saveAndFlush(automatedTask);
+                log.debug("Updated the task {} with cron expression {}", taskName, cronExpression);
+            } else {
+                log.error("Unable to update the task {} with cron expression {}", taskName, cronExpression);
+                throw new DartsApiException(INVALID_CRON_EXPRESSION);
+            }
+        } else {
+            log.error("Failed to update the task {} with cron expression {}", taskName, cronExpression);
+            throw new DartsApiException(FAILED_TO_FIND_AUTOMATED_TASK);
+        }
+        return true;
+    }
+
+    @Override
+    public void reloadTaskByName(String taskName) {
+        switch (AutomatedTaskName.valueOfTaskName(taskName)) {
+            case PROCESS_DAILY_LIST_TASK_NAME -> rescheduleProcessDailyListAutomatedTask();
+            default -> throw new DartsApiException(FAILED_TO_FIND_AUTOMATED_TASK);
+        }
+    }
+
+    @Override
+    public boolean cancelAutomatedTaskAndUpdateCronExpression(String taskName, boolean mayInterruptIfRunning, String cronExpression) {
+        boolean cancelled = cancelAutomatedTask(taskName, mayInterruptIfRunning);
+        boolean updatedCronExpresssion = false;
+        if (cancelled) {
+            updatedCronExpresssion = updateAutomatedTaskCronExpression(taskName, cronExpression);
+            if (updatedCronExpresssion) {
+                reloadTaskByName(taskName);
+            } else {
+                throw new DartsApiException(FAILED_TO_FIND_AUTOMATED_TASK);
```

Comment:
Is this `else` branch is reachable?

Looks like the only possible outcomes of `updateAutomatedTaskCronExpression()` are:
- `true`
-  or an exception is thrown

Reply:
this has changed